validator/client/runner.go

```diff
@@ -110,7 +102,11 @@ func run(ctx context.Context, v iface.Validator) {
 			// Keep trying to update assignments if they are nil or if we are past an
 			// epoch transition in the beacon node's state.
 			if err := v.UpdateDuties(ctx, slot); err != nil {
-				handleAssignmentError(err, slot)
+				if errors.Is(err, ErrValidatorsAllExited) {
```

Comment:
Why not have this under `handleAssignmentError`? It already handles pending validators anyway. It seems cleaner to push this responsibility to `handleAssignmentError` under a case state ment for pending, exit, and everything else

Reply:
I was thinking that initially I'm ok either way i suppose, thought handle assignment error dealt with some other things but i'll move it.